bin/client/src/l1/driver.rs

```diff
@@ -280,6 +287,9 @@ where
                                         .signal(),
                                     )
                                     .await?;
+                            } else if matches!(e, ResetError::ReorgDetected(_, _)) {
+                                self.caching_oracle.as_ref().flush();
+                                self.pipeline.signal(Signal::FlushChannel).await?;
```

Comment:
We will need to perform a full reset on a reorg and not just a channel flush.
Basically, we need to check inside the `else` branch below if `e` matches the `ResetError::ReorgDetected(_, _)`, then flush the caching oracle and resume with sending a reset signal.

e.g.
```rust
  } else {
      if matches!(e, ResetError::ReorgDetected(_, _)) {
            self.caching_oracle.as_ref().flush();
      }
      // Reset the pipeline to the initial L2 safe head and L1 origin,
      // and try again.
      self.pipeline
          .signal(
              ResetSignal {
                  l2_safe_head: self.l2_safe_head,
                  l1_origin: self
                      .pipeline
                      .origin()
                      .ok_or_else(|| anyhow!("Missing L1 origin"))?,
                  system_config: Some(system_config),
              }
              .signal(),
          )
          .await?;
  }
```